lib/matplotlib/patches.py

```diff
@@ -4004,46 +4004,76 @@ def __init__(self, posA=None, posB=None,
                  connector=None,
                  patchA=None,
                  patchB=None,
-                 shrinkA=2.,
-                 shrinkB=2.,
-                 mutation_scale=1.,
+                 shrinkA=2,
+                 shrinkB=2,
+                 mutation_scale=1,
                  mutation_aspect=None,
-                 dpi_cor=1.,
+                 dpi_cor=1,
                  **kwargs):
         """
-        If *posA* and *posB* is given, a path connecting two point are
-        created according to the connectionstyle. The path will be
+        If *posA* and *posB* are given, a path connecting two points are
+        created according to *connectionstyle*. The path will be
         clipped with *patchA* and *patchB* and further shrunken by
         *shrinkA* and *shrinkB*. An arrow is drawn along this
-        resulting path using the *arrowstyle* parameter. If *path*
-        provided, an arrow is drawn along this path and *patchA*,
-        *patchB*, *shrinkA*, and *shrinkB* are ignored.
+        resulting path using the *arrowstyle* parameter.
 
-        The *connectionstyle* describes how *posA* and *posB* are
-        connected. It can be an instance of the ConnectionStyle class
-        (matplotlib.patches.ConnectionStlye) or a string of the
-        connectionstyle name, with optional comma-separated
-        attributes.  The following connection styles are available.
+        Alternatively if *path* is provided, an arrow is drawn along this path
+        and *patchA*, *patchB*, *shrinkA*, and *shrinkB* are ignored.
 
-        %(AvailableConnectorstyles)s
+        Parameters
+        ----------
+        posA, posB: None, tuple, optional (default: None)
+            (x,y) co-ordinates of arrow tail and arrow head respectively.
 
+        path: None, Path (default: None)
+            :class:`matplotlib.path.Path` instance. If provided, an arrow is
+            drawn along this path and *patchA*, *patchB*, *shrinkA*, and
+            *shrinkB* are ignored.
 
-        The *arrowstyle* describes how the fancy arrow will be
-        drawn. It can be string of the available arrowstyle names,
-        with optional comma-separated attributes, or one of the
-        ArrowStyle instance. The optional attributes are meant to be
-        scaled with the *mutation_scale*. The following arrow styles are
-        available.
+        arrowstyle: ArrowStyle or string, optional (default: 'simple')
+            Describes how the fancy arrow will be
+            drawn. It can be string of the available arrowstyle names,
+            with optional comma-separated attributes, or an
+            :class:`ArrowStyle` instance. The optional attributes are meant to
+            be scaled with the *mutation_scale*. The following arrow styles are
+            available:
 
-        %(AvailableArrowstyles)s
+            %(AvailableArrowstyles)s
```

Comment:
We are working towards removing docstring interpolation. If the available arrowstyles are documented else where, can you replace this with the location of the documentation instead of the interpolation? If it isn't documented elsewhere, we can focus on this later.